K8s Helm chart v4 replacement (#3282)
SumoLogic/sumologic-documentation: docs/observability/kubernetes/collection-setup.md

This page provides an overview of the collection process for Kubernetes environments.

> Sumo Logic collects logs, events, metrics, and security data with the [Sumo Logic Distribution for OpenTelemetry Collector](https://github.com/SumoLogic/sumologic-otel-collector), Prometheus, and Falco. These collectors are all open source collectors that are maintained by the Cloud Native Computing Foundation (CNCF). The collected data streams through the OpenTelemetry pipelines for metadata enrichment. Sumo Logic tags the container, pod, node, and cluster, as well as identifying the service, namespace, and deployment.

Author: @swiatekm-sumo @kkujawa-sumo , Does this language need to be updated? Helm v4 doesn't use Prometheus, right?

Reviewer: By default, Helm v4 does not use Prometheus, but it is still possible to enable it.

Reviewer 2: in Helm Chart v4, by default, the OpenTelemetry Collector is now used for metrics collection instead of Prometheus ;) 
I suggest removing "Prometheus" from this section.
 
https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/v4-migration-doc.md#opentelemetry-collector-for-metrics-collection

Author: thank you both!

> Sumo Logic collects logs, events, metrics, and security data with the [Sumo Logic Distribution for OpenTelemetry Collector](https://github.com/SumoLogic/sumologic-otel-collector), Prometheus, and Falco. These collectors are all open source collectors that are maintained by the Cloud Native Computing Foundation (CNCF). The collected data streams through the OpenTelemetry pipelines for metadata enrichment. Sumo Logic tags the container, pod, node, and cluster, as well as identifying the service, namespace, and deployment.

Author: ```suggestion
Sumo Logic collects logs, events, metrics, and security data using the [Sumo Logic Distribution for OpenTelemetry Collector](https://github.com/SumoLogic/sumologic-otel-collector) and Falco, both of which are open source collectors. The collected data streams through the OpenTelemetry pipelines for metadata enrichment. Sumo Logic then tags the container, pod, node, and cluster, as well as identifying the service, namespace, and deployment.
```
